code/game/atoms.dm

```diff
@@ -164,6 +164,22 @@
 	SEND_SIGNAL(src, COMSIG_ATOM_DIR_CHANGE, dir, newdir)
 	dir = newdir
 
+/*
+	Sets the atom's pixel locations based on what pixel offset arguments are passed into it
+	If no arguments are supplied, `pixel_x` or `pixel_y` will be set to 0
+	Used primarily for when players attach mountable frames to walls (APC frame, fire alarm frame, etc.)
+*/
+/atom/proc/set_pixel_offsets(pixel_north = 0, pixel_south = 0, pixel_east = 0, pixel_west = 0)

```

Comment:
This is named much more generically than it actually is. Many things want to set pixel offsets. Only mountable frames want to use this function.
It would be a fine name if it was a method on /obj/item/mounted/frame/ but it isn't (understandably so) - it's a global proc.
Please, please include some kind of "mountable frame" mention in the name
`set_pixel_offsets_for_mountable_frames` ,  `pixel_offsets_apc` or even `apc_pixel_fix` would all work.

Reply:
I honestly can't decide on a name that I like, maybe I *should* just make it a mountable frame only proc.

Reply:
Actually, yeah, you can technically make a new `/obj/item/mounted/frame/proc/set_pixel_offsets` proc, and call that from other places via its full path (i.e. `/obj/item/mounted/frame/proc/set_pixel_offsets(-24, 24, -24, 24)`) (at least i'm 83% sure this is the case) (you might need to pass the atom as an extra argument, since I'm not 100% it would work with `src`)

Reply:
Agghh, actually I can't because all these objects I'm touching have different paths:
1. `/obj/structure/extinguisher_cabinet`
2. `/obj/machinery/defibrillator_mount`
3. `/obj/item/radio/intercom`

For now I think I will go with `set_pixel_offsets_from_dir`. Generic yes, but hopefully you can tell what its suppose to do by its name, and by then reading the documentation for it.

Reply:
no, what I'm saying is you can have

```dm
/obj/item/mounted/frame/proc/set_pixel_offsets(/obj/item/frame, pixel_north = 0, pixel_south = 0, pixel_east = 0, pixel_west = 0)
  switch(frame.dir)
    if (NORTH):
      frame.pixel_y = pixel_north
  // similarly other cases

// and then, later, in a different dm:

/obj/machinery/alarm/New(loc, direction, building = 0)
  // ..
  if(building):
    /obj/item/mounted/frame/proc/set_pixel_offsets(src, -24, 24, -24, 24)
```

Again, I'm only ~80% sure that this is the correct syntax (can't test rn :c ), but I think it should work.

Reply:
I don't think that's even posssible to call procs like that, and even if it was, I would get stabbed for trying :smile:. 

But see the issue is that I can't create a single proc for all the mountable frames becuase, as I wrote in my previous comment, a lot of them have quite different paths, (one is `/obj/item`, one is `/obj/machinery`, etc.). Which is why I just went down the `/atom` level. But now that I think about it, I could also just do it on the `/obj` level.